### PR TITLE
Update configurations to run on 2016 and 2017 eras.

### PR DESCRIPTION
**Pull request by mburkart**

### CODE REVIEW

File: config.py [hunk 1]
```diff
@@ -511,19 +529,23 @@ def build_config(
...
+                    "2016preVFP": "data/recoil_corrections/PuppiMETSys_2016.root",  # These are likely from Legacy data sets, therefore no difference in pre and postVFP
+                    "2016postVFP": "data/recoil_corrections/PuppiMETSys_2016.root",  # These are likely from Legacy data sets, therefore no difference in pre and postVFP
                     "2017": "data/recoil_corrections/PuppiMETSys_2017.root",
                     "2018": "data/recoil_corrections/PuppiMETSys_2018.root",
                 }
             ),
-            "applyRecoilCorrections": SampleModifier({"wjets": True}, default=False),
+            "applyRecoilCorrections": SampleModifier(
+                {"wjets": True}, default=False
+            ),  # ToDo: Shouldn't this be also applied to dy and higgs events?
```
Comment (harrypuuter) [inline config.py:548]: good question :D

File: generate.py [hunk 2]
```diff
@@ -23,7 +23,7 @@ def run(args):
         "data",
         "electroweak_boson",
     ]
-    available_eras = ["2016", "2017", "2018"]
+    available_eras = ["2016preVFP", "2016postVFP", "2017", "2018"]
```
Comment (harrypuuter) [inline generate.py:26]: also for generate_friends.py

File: tau_triggersetup.py [hunk 3]
```diff
@@ -74,7 +103,7 @@ def add_diTauTriggerSetup(configuration: Configuration):
                             "flagname": "trg_cross_mu20tau27_hps",
                             "hlt_path": "HLT_IsoMu20_eta2p1_LooseChargedIsoPFTauHPS27_eta2p1_CrossL1",
                             "p1_ptcut": 21,
-                            "p1_etacut": 2.5,
+                            "p1_etacut": 2.5,  # ToDo: Shouldn't this be 2.1 as well?
```
Comment (harrypuuter) [inline tau_triggersetup.py:106]: should be 2.1

**Review by harrypuuter — CHANGES_REQUESTED** [review]
small changes

File: config.py (hunk 1 above)
Comment (mburkart) [reply config.py:548]: At least the pipelines have been used like this in the past:
https://github.com/KIT-CMS/KITHiggsToTauTau/blob/reduced_trigger_objects/python/data/ArtusConfigs/Run2LegacyAnalysis/METrecoil_shifts.py#L39

Comment (harrypuuter) [reply config.py:548]: then, lets add it

File: tau_embedding_settings.py [hunk 4]
```diff
@@ -756,6 +847,21 @@ def setup_embedding(configuration: Configuration, scopes: List[str]):
                 samples=["embedding"],
             )
     else:
+        tauES_2016preVFP = {
```
[the remark this answers is not in the document]
Comment (mburkart) [reply tau_embedding_settings.py]: Added.

File: tau_triggersetup.py (hunk 3 above)
Comment (mburkart) [reply tau_triggersetup.py]: Fixed.

**Review by harrypuuter — APPROVED** [review]

File: config.py (hunk 1 above)
Comment (mburkart) [reply config.py]: Added.

**Comment by harrypuuter** [comment]
lets merge ?

**Pull request merged**